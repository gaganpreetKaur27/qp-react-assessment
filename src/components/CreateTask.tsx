import React, { useCallback, useState } from "react";
//utils
import { CreateTaskProp } from "../utils/types/TaskTypes";
import "../utils/css/styles.css";

export const CreateTask: React.FC<CreateTaskProp> = ({ addTask }) => {
  const [value, setValue] = useState("");

  const handleAddTask = useCallback(() => {
    if (!value) return;
    addTask(value);
    setValue("");
  },[value]);

  return (
  <div className="add-task-container">
      <input
        type="text"
        className="input"
        value={value}
        placeholder="Add a new task to the list.. :)"
        onChange={(e) => setValue(e.target.value)}
      />
      <button className="add-button-style" onClick={handleAddTask}>
       Add Task
    </button>
    </div>
  );
};
