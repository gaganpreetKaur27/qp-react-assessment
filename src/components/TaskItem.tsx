import React from "react";
//utils
import { TaskProps } from "../utils/types/TaskTypes";
import "../utils/css/styles.css";

export const TaskItem: React.FC<TaskProps> = ({
  task,
  index,
  completeTask,
  removeTask,
}) => {
  return (
    <div
      className="task"
      style={{ textDecoration: task.completed ? "line-through" : "" }}
      key={index}
    >
      <span>{task.title}</span>
      <button
        className="button-container"
        style={{ background: "red" }}
        onClick={() => removeTask(index)}
      >
        x
      </button>
      <button
        data-testID="complete-btn"
        className={
          !task.completed ? "checkbox-button-pending" : "button-container"
        }
        onClick={() => completeTask(index)}
      >
        {task.completed ? "Completed" : ""}
      </button>
    </div>
  );
};
