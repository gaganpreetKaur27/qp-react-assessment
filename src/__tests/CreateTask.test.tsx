import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { CreateTask } from "../components/CreateTask";

describe("CreateTask component", () => {
  const addTaskMock = jest.fn();

  it("CreateTask matches snapshot", () => {
    const { asFragment } = render(<CreateTask addTask={addTaskMock} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("CreateTask update input field value", () => {
    const { getByPlaceholderText } = render(
      <CreateTask addTask={addTaskMock} />
    );
    const inputField = getByPlaceholderText(
      "Add a new task to the list.. :)"
    ) as HTMLInputElement;
    // Act
    fireEvent.change(inputField, { target: { value: "New task" } });
    // Assert
    expect(inputField.value).toBe("New task");
  });
});
