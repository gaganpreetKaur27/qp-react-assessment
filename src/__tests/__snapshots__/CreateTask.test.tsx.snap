// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateTask component CreateTask matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="add-task-container"
  >
    <input
      class="input"
      placeholder="Add a new task to the list.. :)"
      type="text"
      value=""
    />
    <button
      class="add-button-style"
    >
      Add Task
    </button>
  </div>
</DocumentFragment>
`;
